### PR TITLE
Initialize Application

### PR DESCRIPTION
**Pull request by messerc**
Screen Shot 2019-09-05 at 4 00 09 PM

- Utilize our internal library [poco](https://github.com/globality-corp/poco) to bootstrap the FE application. Poco primarily takes care of adding services that the application will need via React `Providers` (pattern described [here](https://www.robinwieruch.de/react-provider-pattern-context)). This beings with the most basic providers we will need.

- The webpack magic is taken care of by another internal library [gi-webapp-build-tools](https://github.com/globality-corp/gi-webapp-build-tools). It handles the webpack configuration as well as other mundane setup for: tests, linting, etc.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Poco, { HistoryPlugin, IntlPlugin, ReduxPlugin } from '@globality/poco';
+
+import createApp from './modules/app';
+
+const App = new Poco({
```
Comment (messerc) [inline src/main.js]: App initialization happening here.

- The Poco class declared [here](https://github.com/globality-corp/poco/blob/develop/src/index.js#L18)

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    appName: 'poco-sample-frontend',
+    createApp,
```
Comment (messerc) [inline src/main.js]: The function that `poco` needs to actually initialize the application

File: src/main.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+});
+
+App.load((app) => {
```
Comment (messerc) [inline src/main.js]: Once the application is initialized, load in the `Providers`, in this case we call them a plugin.

File: src/modules/app/index.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+const SecondPage = () => {
+    return (
+        <div>
+            <p>Secondary page</p>
+            <Link to='/'>Go back</Link>
+        </div>
+    );
+}
+
+export default function createApp ({ container, onReady }) {
```
Comment (messerc) [inline src/modules/app/index.js]: The root of the application. The `onReady` is a convenience function for when the application is done doing any initial loading.

Poco takes in this `createApp` function, [does some nice things and renders it](https://github.com/globality-corp/poco/blob/0ceb2cd60d3298ff58f49575affe0aa94a8ddcd9/src/index.js#L116)

File: src/modules/app/index.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    const { history } = container;
+    setTimeout(() => onReady(), 1000);
+
+    return (
+        <Router history={history}>
```
Comment (messerc) [inline src/modules/app/index.js]: `react-router` [v4](https://reacttraining.com/react-router/web/guides/quick-start) used in all of our FE applications

File: package.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "poco-sample-frontend",
+  "version": "0.0.1",
+  "description": "A simple front-end example application built using the Poco framework. This repo is designed as a guide to help when on-boarding new engineers at Globality.",
+  "author": "Globality Engineering",
+  "license": "UNLICENSED",
+  "private": true,
+  "contributors": [
+    {
+      "name": "Tom Hill",
```
Comment (cdm-ium) [inline package.json]: Let's swap this out with globality engineering email.

**Pull request merged**